judo-runtime-core-jsl-itest/common/src/main/java/hu/blackbelt/judo/runtime/core/jsl/fixture/JudoDatasourceFixture.java

```diff
@@ -95,24 +102,43 @@ public void teardownDatasource() {
         }
     }
 
-    public void dropSchema() {
+    public void truncateTables(RdbmsModel rdbmsModel) {
+        RdbmsUtils rdbmsUtils = new RdbmsUtils(rdbmsModel.getResourceSet());
         try (Connection connection = dataSource.getConnection(); Statement statement = connection.createStatement()) {
             if (dialect.equals(DIALECT_POSTGRESQL)) {
-                statement.execute("select 'drop table \"' || tablename || '\" cascade;' from pg_tables;");
+                for (RdbmsTable rdbmsTable : rdbmsUtils.getRdbmsTables().orElse(new BasicEList<>())) {
+                    log.debug("Truncating table: %s (%s)".formatted(rdbmsTable.getName(), rdbmsTable.getSqlName()));
+                    statement.execute("TRUNCATE TABLE " + rdbmsTable.getSqlName() + " RESTART IDENTITY CASCADE;");
+                }
             } else if (dialect.equals(DIALECT_HSQLDB)) {
-                statement.execute("TRUNCATE SCHEMA PUBLIC RESTART IDENTITY AND COMMIT NO CHECK");
-                //statement.execute("DROP SCHEMA PUBLIC CASCADE");
+                for (RdbmsTable rdbmsTable : rdbmsUtils.getRdbmsTables().orElse(new BasicEList<>())) {
+                    log.info("Truncating table: %s (%s)".formatted(rdbmsTable.getName(), rdbmsTable.getSqlName()));
+                    statement.execute("TRUNCATE TABLE " + rdbmsTable.getSqlName() + " RESTART IDENTITY AND COMMIT NO CHECK");
+                }
             }
         } catch (SQLException throwables) {
-            throw new RuntimeException("Could not drop schema", throwables);
+            throw new RuntimeException("Could not truncate tables", throwables);
         }
+    }
 
+    public void dropTables(RdbmsModel rdbmsModel) {
+        RdbmsUtils rdbmsUtils = new RdbmsUtils(rdbmsModel.getResourceSet());
+        try (Connection connection = dataSource.getConnection(); Statement statement = connection.createStatement()) {
+            if (dialect.equals(DIALECT_POSTGRESQL) || dialect.equals(DIALECT_HSQLDB)) {
+                for (RdbmsTable rdbmsTable : rdbmsUtils.getRdbmsTables().orElse(new BasicEList<>())) {
+                    log.debug("Drop table: %s (%s)".formatted(rdbmsTable.getName(), rdbmsTable.getSqlName()));
+                    statement.execute("DROP TABLE " + rdbmsTable.getSqlName() + " CASCADE;");
+                }
+            }
+        } catch (SQLException throwables) {
+            throw new RuntimeException("Could not drop tables", throwables);
+        }
```

Comment:
Whu the dialect selection here? There is no other alternative.

Reply:
[01b3021](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/242/commits/01b30214a4bec63182a30c9527cdd8ca8b092286)